com.hazelcast.mapstore.mysql.MySQLGenericMapStoreIntegrationTest.testDestroy [HZ-2746]

_master_ (commit 98988dcd386ad8c413ec9cb061a714dd5dac51a8)

Failed on openjdk-sts (openjdk-20): https://jenkins.hazelcast.com/job/Hazelcast-master-OpenJDK-sts-nightly/36/testReport/com.hazelcast.mapstore.mysql/MySQLGenericMapStoreIntegrationTest/testDestroy/
<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: 

Expecting
  [Row{[__map-store.mysql_testdestroy]},
    Row{[__map-store.mysql_testget]},
    Row{[__map-store.mysql_testloadallkeys_disabled]},
    Row{[__map-store.mysql_testputwhenexists]},
    Row{[__map-store.mysql_testputwithgenericrecordidcolumnignored]}]
not to contain
  [Row{[__map-store.mysql_testdestroy]}]
but found
  [Row{[__map-store.mysql_testdestroy]}]

	at com.hazelcast.jet.sql.SqlTestSupport.assertDoesNotContainRow(SqlTestSupport.java:408)
	at com.hazelcast.mapstore.GenericMapStoreIntegrationTest.lambda$testDestroy$3(GenericMapStoreIntegrationTest.java:392)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1165)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1184)
	at com.hazelcast.mapstore.GenericMapStoreIntegrationTest.testDestroy(GenericMapStoreIntegrationTest.java:392)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
	at java.base/java.lang.reflect.Method.invoke(Method.java:578)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.lang.Thread.run(Thread.java:1623)

```
</details>


<details><summary>Standard output:</summary>

```
Finished Running Test: testLoadAllKeys in 0.112 seconds.
Started Running Test: testDestroy
20:33:15,282 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-28 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testdestroy
20:33:15,282  INFO || - [DefaultRecordStore] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.partition-operation.thread-3 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Starting to load all keys for map mysql_testdestroy on partitionId=195
20:33:15,325 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-8 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testdestroy
20:33:15,341 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-8 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Reading existing mapping for mapmysql_testdestroy
20:33:17,356  INFO || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-54 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Dropping mapping __map-store.mysql_testdestroy
20:33:17,357  INFO || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-26 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Dropping mapping __map-store.mysql_testdestroy
20:33:17,358 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-32 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testdestroy
20:33:19,635 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-54 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testget
20:33:19,684 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-51 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testget
20:33:19,708 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-51 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Reading existing mapping for mapmysql_testget
20:33:19,837 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-47 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testputwithgenericrecordidcolumnignored
20:33:19,874 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-46 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testputwithgenericrecordidcolumnignored
20:33:19,903 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-46 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Reading existing mapping for mapmysql_testputwithgenericrecordidcolumnignored
20:33:20,030 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-46 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testputwhenexists
20:33:20,072 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-42 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testputwhenexists
20:33:20,092 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-42 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Reading existing mapping for mapmysql_testputwhenexists
20:33:20,176 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-46 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testloadallkeys_disabled
20:33:20,177 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_lucid_booth.cached.thread-13 - [127.0.0.1]:5702 [dev] [5.4.0-SNAPSHOT] Initializing for map mysql_testloadallkeys_disabled
20:33:20,204 DEBUG || - [GenericMapLoader] hz.MySQLGenericMapStoreIntegrationTest_serene_booth.cached.thread-46 - [127.0.0.1]:5701 [dev] [5.4.0-SNAPSHOT] Reading existing mapping for mapmysql_testloadallkeys_disabled
20:33:49,525  INFO |testDestroy| - [SqlTestSupport] Time-limited test - Removing 3 cached plans in SqlTestSupport.@After
20:33:49,526  INFO |testDestroy| - [SqlTestSupport] Time-limited test - Removing 4 cached plans in SqlTestSupport.@After
20:33:49,528  INFO |testDestroy| - [SimpleTestInClusterSupport] Time-limited test - Ditching 0 jobs in SimpleTestInClusterSupport.@After: []
20:33:49,528  INFO |testDestroy| - [SimpleTestInClusterSupport] Time-limited test - Destroying 1 distributed objects in SimpleTestInClusterSupport.@After: [hz:impl:mapService/__sql.catalog]
20:33:49,530  INFO |testDestroy| - [HazelcastTestSupport] Time-limited test - Shutting down node factory as @After action
BuildInfo right after testDestroy(com.hazelcast.mapstore.mysql.MySQLGenericMapStoreIntegrationTest): BuildInfo{version='5.4.0-SNAPSHOT', build='20230701', buildNumber=20230701, revision=98988dc, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'testDestroy(com.hazelcast.mapstore.mysql.MySQLGenericMapStoreIntegrationTest):'
20:33:15, accumulated pauses: 50 ms, max pause: 1 ms, pauses over 1000 ms: 0
20:33:20, accumulated pauses: 52 ms, max pause: 2 ms, pauses over 1000 ms: 0
20:33:25, accumulated pauses: 51 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:33:30, accumulated pauses: 51 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:33:35, accumulated pauses: 64 ms, max pause: 10 ms, pauses over 1000 ms: 0
20:33:40, accumulated pauses: 51 ms, max pause: 0 ms, pauses over 1000 ms: 0
20:33:45, accumulated pauses: 1949 ms, max pause: 1923 ms, pauses over 1000 ms: 1


No metrics recorded during the test

```
</details>

Standard output can be found here - https://s3.console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-master-OpenJDK-sts-nightly/36/

--- a/hazelcast/src/main/java/com/hazelcast/map/impl/LocalMapStatsProvider.java
+++ b/hazelcast/src/main/java/com/hazelcast/map/impl/LocalMapStatsProvider.java
@@ -350,7 +350,10 @@ public class LocalMapStatsProvider {
     }
 
     private void addIndexStats(String mapName, LocalMapStatsImpl localMapStats) {
-        MapContainer mapContainer = mapServiceContext.getMapContainer(mapName);
+        MapContainer mapContainer = mapServiceContext.getExistingMapContainer(mapName);
+        if (mapContainer == null) {
+            return;
+        }
         Indexes globalIndexes = mapContainer.getIndexes();
 
         Map<String, OnDemandIndexStats> freshStats = null;

